src/data/nat/choose.lean

```diff
@@ -9,13 +9,23 @@ open nat
 
 open_locale big_operators
 
-lemma nat.prime.dvd_choose {p k : ℕ} (hk : 0 < k) (hkp : k < p) (hp : prime p) : p ∣ choose p k :=
-have h₁ : p ∣ fact p, from hp.dvd_fact.2 (le_refl _),
-have h₂ : ¬p ∣ fact k, from mt hp.dvd_fact.1 (not_le_of_gt hkp),
-have h₃ : ¬p ∣ fact (p - k), from mt hp.dvd_fact.1 (not_le_of_gt (nat.sub_lt_self hp.pos hk)),
-by rw [← choose_mul_fact_mul_fact (le_of_lt hkp), mul_assoc, hp.dvd_mul, hp.dvd_mul] at h₁;
+lemma nat.prime.dvd_choose {p a b : ℕ} (hap : a < p) (hbp : b < p) (h : p ≤ a + b)
+  (hp : prime p) : p ∣ choose (a + b) a :=
+have h₁ : p ∣ fact (a + b), from hp.dvd_fact.2 h,
+have h₂ : ¬p ∣ fact a, from mt hp.dvd_fact.1 (not_le_of_gt hap),
+have h₃ : ¬p ∣ fact b, from mt hp.dvd_fact.1 (not_le_of_gt hbp),
+by
+  rw [← choose_mul_fact_mul_fact (le.intro rfl), mul_assoc, hp.dvd_mul, hp.dvd_mul,
+    norm_num.sub_nat_pos (a + b) a b rfl] at h₁;
   exact h₁.resolve_right (not_or_distrib.2 ⟨h₂, h₃⟩)
 
+lemma nat.prime.dvd_choose_self {p k : ℕ} (hk : 0 < k) (hkp : k < p) (hp : prime p) : p ∣ choose p k :=
+begin
+  have r : k + (p - k) = p, by omega,
```

Comment:
I think `omega` is super slow. At any rate, it's not mathlib style to use it on trivial goals that could be solved by a term mode proof. `library_search` should tell you what you need here, and below.